### PR TITLE
Add agentic competition tasks, agent definitions, and skills

### PR DESCRIPTION
**Pull request by AaronGoldsmith**
## Summary
- 3 new agent definitions (competition-tasks, depth-test, tree-solver)
- 2 new skills (/mobius-evolve, /tree-solve)
- 25 general + 15 agentic competition tasks across 3 difficulty tiers
- Cleanup script for dead weight agents

## Fixes from PR #10 review
- Removed hardcoded absolute paths (now relative)
- Made tasks cross-platform (replaced Windows-only commands)
- Removed unused import in cleanup script

## Test plan
- [ ] Verify JSON task files parse correctly
- [ ] Skill files load without errors
- [ ] No hardcoded user-specific paths

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by AaronGoldsmith** [comment]
Splitting into smaller, focused PRs.

File: .claude/skills/mobius-evolve/scripts/load_underperformers.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+so that Opus can craft improved system prompts.
+"""
+
+import json
+import sys
+
+sys.path.insert(0, "src")
+
+from mobius.config import get_config
+from mobius.db import init_db, row_to_dict
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply .claude/skills/mobius-evolve/scripts/load_underperformers.py:16]: Addressed — removed unused json and row_to_dict imports.

File: scripts/competition_tasks.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
+[
+  {
+    "task": "Implement a rate limiter using the token bucket algorithm in Python with thread safety. Include a clean API: `RateLimiter(rate, capacity)` with an `acquire()` method that blocks until a token is available and a `try_acquire()` that returns immediately.",
+    "category": "code-generation-python"
+  },
+  {
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply scripts/competition_tasks.json:6]: Addressed — task loader now tries json.loads() first (extracting 'task' fields from objects), falling back to splitlines() for plain text.

File: scripts/competition_tasks_agentic.json [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    "category": "explore-and-analyze",
+    "tier": 3,
+    "tools_required": [
+      "Bash"
+    ],
+    "note": "Designed for future multi-agent mode (Agent A audits, Agent B validates and reproduces). Currently runs as single-agent doing both roles.",
+    "verification": "Judge checks: (1) error_audit.json has 5+ findings with file:line references, (2) validation.json marks each as confirmed/false-positive with evidence, (3) reproduction scripts actually demonstrate the issues, (4) proposed fixes are reasonable and don't break existing functionality",
+    "setup": "mkdir -p /tmp/mobius-task-errors"
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply scripts/competition_tasks_agentic.json:144]: Addressed — changed `at ..` to `at .` in this task as well.

File: scripts/competition_tasks_agentic.json [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  {
+    "task": "Create a Python CLI tool at /tmp/mobius-task-csvtool/csvtool.py that reads a CSV file, supports --filter COLUMN=VALUE, --sort COLUMN, --group-by COLUMN (with count aggregation), and --output FORMAT (table or json). Generate a test CSV with at least 100 rows of synthetic employee data (name, department, salary, hire_date). Run the tool with each flag combination and verify the output is correct. Fix any bugs you find.",
+    "category": "build-and-test",
+    "tier": 1,
+    "tools_required": [
+      "Bash"
+    ],
+    "verification": "Judge checks: (1) csvtool.py exists and is executable, (2) test CSV has 100+ rows, (3) all four flag combinations produce correct output when re-run, (4) edge cases like missing columns produce helpful errors",
+    "setup": "mkdir -p /tmp/mobius-task-csvtool"
+  },
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply scripts/competition_tasks_agentic.json:11]: Won't do — /tmp paths are intentional for task isolation. These tasks create throwaway artifacts that shouldn't live in the repo. Cross-platform support can be addressed separately if needed.

File: scripts/cleanup_agents.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+    if not args.execute:
+        print("[DRY-RUN] No changes made. Re-run with --execute to apply.")
+        print(f"  Would retire {len(zero_match)} zero-match agents (set elo=0, is_champion=0)")
+        print(f"  Would clear is_champion flag on {len(champions)} agents")
+    else:
+        retired = retire_zero_match_agents(conn)
+        cleared = clear_all_champion_flags(conn)
+        print(f"[EXECUTE] Retired {retired} zero-match agents (elo set to 0)")
+        print(f"[EXECUTE] Cleared champion flag on {cleared} agents")
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply scripts/cleanup_agents.py:108]: Addressed — added elect_champions() that re-elects highest-Elo agent per specialization after clearing flags.

File: scripts/competition_tasks_agentic.json [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  {
+    "task": "Reverse-engineer the Mobius agent selection algorithm by reading the source code at .. Then write a simulation at /tmp/mobius-task-selection/simulate.py that models 100 competitions with 10 agents starting at Elo 1000. Track how Elo ratings drift, identify whether the selection algorithm has any bias (does it favor certain agents unfairly?), and determine after how many rounds the rankings stabilize. Output a report with the Elo trajectory data and your analysis to report.txt. The simulation should use the same Elo update formula as the real code.",
+    "category": "explore-and-analyze",
+    "tier": 2,
+    "tools_required": [
+      "Bash"
+    ],
+    "verification": "Judge checks: (1) simulation uses the actual Elo formula from Mobius source (not a generic one), (2) simulate.py runs and produces trajectory data, (3) report.txt contains analysis of bias and stabilization with supporting data, (4) at least one non-obvious insight about the selection algorithm",
+    "setup": "mkdir -p /tmp/mobius-task-selection"
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply scripts/competition_tasks_agentic.json:100]: Addressed — changed `at ..` to `at .` in affected tasks.

File: .claude/skills/mobius-evolve/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
+---
+name: mobius-evolve
+description: Use when the user says "evolve", "mobius evolve", "improve agents", or wants to refine underperforming agents without API costs.
+user-invocable: true
+argument-hint: "[specialization] [--threshold 0.4]"
+---
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply .claude/skills/mobius-evolve/SKILL.md:6]: Addressed — updated argument-hint to include `[--min-matches N]`.

File: .claude/agents/tree-solver.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+1. Write a plan.md to {TREE_WORKSPACE}/{TREE_NODE}/
+2. Create 2-4 child task files at {TREE_WORKSPACE}/{TREE_NODE}-N/task.md
+3. Spawn each child with: `claude -p "$(cat {path}/task.md)" --agent tree-solver --max-turns 20 --allowedTools "Bash,Read,Write,Edit,Grep,Glob" > {path}/output.log 2>&1`
+4. Use `&` and `wait` for independent children
+5. After all finish, read their result.md files, write your own aggregated result.md
+
+**IF TREE_DEPTH == TREE_MAX_DEPTH:**
+You MUST spawn 2 competing experts, NOT do the work yourself:
+1. `claude -p "{expert prompt with approach A}" --model haiku --max-turns 15 --allowedTools "Bash,Read,Write,Edit,Grep,Glob" > {TREE_WORKSPACE}/{TREE_NODE}/expert-1.log 2>&1 &`
+2. `claude -p "{expert prompt with approach B}" --model haiku --max-turns 15 --allowedTools "Bash,Read,Write,Edit,Grep,Glob" > {TREE_WORKSPACE}/{TREE_NODE}/expert-2.log 2>&1 &`
```
[the remark this answers is not in the document]
Comment (AaronGoldsmith) [reply .claude/agents/tree-solver.md:26]: Addressed — removed Write and Edit from allowedTools in all spawn commands, now consistent with the 'Do NOT use Write' instruction.
